esphome/components/hbridge/__init__.py

```diff
@@ -1,3 +1,62 @@
 import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import output
+from esphome.const import CONF_PIN_A, CONF_PIN_B, CONF_ENABLE_PIN
 
 hbridge_ns = cg.esphome_ns.namespace("hbridge")
+
+CONF_TRANSITION_DELTA_PER_MS = "transition_delta_per_ms"
+CONF_TRANSITION_SHORT_BUILDUP_DURATION = "transition_short_buildup_duration"
+CONF_TRANSITION_FULL_SHORT_DURATION = "transition_full_short_duration"
+
+HBRIDGE_CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_PIN_A): cv.use_id(output.FloatOutput),
+        cv.Required(CONF_PIN_B): cv.use_id(output.FloatOutput),
+        cv.Optional(CONF_ENABLE_PIN): cv.use_id(output.FloatOutput),
+        cv.Optional(CONF_TRANSITION_DELTA_PER_MS, default="2"): cv.float_range(
+            min=0, min_included=False
+        ),
+        cv.Optional(
+            CONF_TRANSITION_SHORT_BUILDUP_DURATION, default="0ms"
+        ): cv.positive_time_period_milliseconds,
+        cv.Optional(
+            CONF_TRANSITION_FULL_SHORT_DURATION, default="0ms"
+        ): cv.positive_time_period_milliseconds,
+    }
+)
+
+
+async def hbridge_config_to_code(config, var):
```

Comment:
The usual convention is:
```suggestion
async def setup_hbridge(var, config):
```

Reply:
I used a different function name on purpose as the hbridge class cannot be used directly and this was more of an "helper function"

Reply:
Yes, that's true, but e.g. the `Climate` or `Fan` class also can't be used on it's own, and they do follow the `setup_` pattern. It's not really important, just a nice bit of consistency :)